### PR TITLE
Fixed #33136 -- Added GEOSGeometry.make_valid() method.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: docs/ref/contrib/gis/geos.txt [hunk 1]
```diff
@@ -511,6 +511,16 @@ Topological Methods
...
 
+.. method:: GEOSGeometry.make_valid()
+
+    .. versionadded::4.1
+
+    Returns a valid :class:`GEOSGeometry` equivalent, trying not to lose any of
+    the input vertices. If the geometry is already valid, it is returned
+    untouched. This is similar to the
+    :class:`~django.contrib.gis.db.models.functions.MakeValid` database
+    function.
```
Comment (felixxm) [inline docs/ref/contrib/gis/geos.txt:522]: Shouldn't this be documented in the `Other Properties & Methods` section? as `normalize()` :thinking:

**Comment by claudep** [comment]
Looks like MakeValid was added in GEOS 3.8, I'll have to check that.

File: docs/ref/contrib/gis/geos.txt (hunk 1 above)
Comment (claudep) [reply docs/ref/contrib/gis/geos.txt:522]: Sure most probably!

File: docs/ref/contrib/gis/geos.txt [hunk 2]
```diff
@@ -511,6 +511,16 @@ Topological Methods
     Returns a :class:`GEOSGeometry` representing the points shared by this
     geometry and other.
 
+.. method:: GEOSGeometry.make_valid()
+
+    .. versionadded::4.1
```
Comment (ngnpope) [inline docs/ref/contrib/gis/geos.txt:516]:
```suggestion
    .. versionadded:: 4.1
```

File: django/contrib/gis/geos/prototypes/geom.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ class StringFromGeom(GEOSFuncFactory):
 
 # The GEOS geometry type, typeid, num_coordinates and number of geometries
 geos_normalize = IntFromGeom('GEOSNormalize')
+geos_makevalid = GeomOutput('GEOSMakeValid', argtypes=[GEOM_PTR])
```
Comment (ngnpope) [inline django/contrib/gis/geos/prototypes/geom.py:48]: This block looks ordered alphabetically (although that may be a fluke when compared to the blocks below):
```suggestion
geos_makevalid = GeomOutput('GEOSMakeValid', argtypes=[GEOM_PTR])
geos_normalize = IntFromGeom('GEOSNormalize')
```

File: docs/ref/contrib/gis/geos.txt (hunk 2 above)
Comment (claudep) [reply docs/ref/contrib/gis/geos.txt]: Thanks, done in the new paragraph location.

File: django/contrib/gis/geos/prototypes/geom.py (hunk 3 above)
Comment (claudep) [reply django/contrib/gis/geos/prototypes/geom.py]: Makes sense!

**Comment by claudep** [comment]
buildbot, test this please.

**Comment by claudep** [comment]
@felixxm Do you think it would be a good idea to run pull requests tests for Python 3.9 on Ubuntu 20.04? Typically for Gis tests at least, this would mean running tests on at least two different Gis lib sets.

**Comment by felixxm** [comment]
@claudep Thanks :+1: I added `test_make_valid_geos_version()` and pushed small edits to release notes.

> @felixxm Do you think it would be a good idea to run pull requests tests for Python 3.9 on Ubuntu 20.04? Typically for Gis tests at least, this would mean running tests on at least two different Gis lib sets.

This would require separate nodes with Ubuntu 20.04 on Jenkins :| I'll think about that.

**Comment by claudep** [comment]
Thanks!

**Pull request merged**